### PR TITLE
[Python] Add execution test for connect with data source

### PR DESCRIPTION
**Pull request by tonyyang-svail**

### CODE REVIEW

**Comment by tonyyang-svail** [comment]
While I am connecting Hive with Auth, I found the following error.

```
ERROR: test_hive (sqlflow_submitter.db_test.TestDB)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/go/src/github.com/sql-machine-learning/sqlflow/sql/python/sqlflow_submitter/db_test.py", line 83, in test_hive
    conn = connect_with_data_source("hive://root:root@127.0.0.1:10000/iris?auth=PLAIN&session.mapreduce_job_quenename=mr")
  File "/go/src/github.com/sql-machine-learning/sqlflow/sql/python/sqlflow_submitter/db.py", line 92, in connect_with_data_source
    auth_mechanism=auth)
  File "/miniconda/envs/sqlflow-dev/lib/python3.6/site-packages/impala/dbapi.py", line 147, in connect
    auth_mechanism=auth_mechanism, krb_host=krb_host)
  File "/miniconda/envs/sqlflow-dev/lib/python3.6/site-packages/impala/hiveserver2.py", line 778, in connect
    auth_mechanism, user, password)
  File "/miniconda/envs/sqlflow-dev/lib/python3.6/site-packages/impala/_thrift_api.py", line 154, in get_transport
    from thrift_sasl import TSaslClientTransport
ModuleNotFoundError: No module named 'thrift_sasl'
```

After Googling, I found we need 
- `sudo apt-get install libsasl2-dev`. This library is required for compiling the `thrift_sasl` package. [ref](https://stackoverflow.com/questions/48562383/sasl-saslwrapper-h2223-fatal-error-sasl-sasl-h-no-such-file-or-directory)
- `pip install thrift_sasl==0.2.1`. [ref](https://github.com/cloudera/impyla/issues/204), [ref](https://github.com/cloudera/impyla/blob/71212b6f5a51e386f8a998f45240144203391ef3/setup.py#L48)

To support testing on auth, we also need to enable auth by turning on `WITH_HS2_PAM_AUTH=ON` at the server-side, as demonstrated in https://github.com/sql-machine-learning/gohive/pull/52.

My question is, the Hive Python connector is currently not tested on Auth. Should we add the test? Will these tests helps to smooth the deployment process? @BlackPoint-CX @Yancey1989

**Comment by Yancey0623** [comment]
> My question is, the Hive Python connector is currently not tested on Auth. Should we add the test? Will these tests helps to smooth the deployment process?

Good question, `thrift_sasl` is necessary if the auth method is not `NOSASAL`,  I think adding a unit test with `WITH_HS2_PAM_AUTH=ON ` is better, I created an issue https://github.com/sql-machine-learning/sqlflow/issues/841.

**Review by Yancey0623 — DISMISSED** [review]

File: sql/python/sqlflow_submitter/db_test.py [hunk 1]
```diff
@@ -172,13 +174,13 @@ def test_parse_mysql_dsn(self):
 
     def test_parse_hive_dsn(self):
         self.assertEqual(
-                ("usr", "pswd", "hiveserver", "1000", "mydb", {"auth":"PLAIN", "session.mapreduce_job_quenename": "mr"}),
+                ("usr", "pswd", "hiveserver", "1000", "mydb", "PLAIN", {"mapreduce_job_quenename": "mr"}),
             parseHiveDSN("usr:pswd@hiveserver:1000/mydb?auth=PLAIN&session.mapreduce_job_quenename=mr"))
         self.assertEqual(
-            ("root", "root", "127.0.0.1", None, "mnist", {"auth":"PLAIN"}),
+            ("root", "root", "127.0.0.1", None, "mnist", "PLAIN", {}),
```
Comment (Yancey0623) [inline sql/python/sqlflow_submitter/db_test.py:180]: ~Seems I don't find the **exception** test?~

**Review by typhoonzero — COMMENTED** [review]
LGTM generally

**Review by Yancey0623 — APPROVED** [review]
LGTM

**Pull request merged**